### PR TITLE
Asserting `every` domain is an iterable type before evaluation

### PR DESCRIPTION
**Pull request by johanfylling**
Fixes: #6762

### CODE REVIEW

File: ast/term.go [hunk 1]
```diff
@@ -556,6 +556,15 @@ func IsScalar(v Value) bool {
 	return false
 }
 
+// IsIterable returns true if the AST value is an iterable type.
+func IsIterable(x Value) bool {
```
Comment (srenatus) [inline ast/term.go:560]: Do we really need to expose this? 🤔 I'd think it work make for a decent helper method in topdown/eval.go, too.

File: topdown/eval.go [hunk 2]
```diff
@@ -3390,19 +3384,32 @@ func (e evalTerm) save(iter unifyIterator) error {
...
 
 func (e evalEvery) eval(iter unifyIterator) error {
 	// unknowns in domain or body: save the expression, PE its body
-	if e.e.unknown(e.generator, e.e.bindings) || e.e.unknown(e.body, e.e.bindings) {
+	if e.e.unknown(e.Domain, e.e.bindings) || e.e.unknown(e.Body, e.e.bindings) {
 		return e.save(iter)
 	}
 
-	domain := e.e.closure(e.generator)
+	if pd := e.e.bindings.Plug(e.Domain); pd != nil {
```
Comment (srenatus) [inline topdown/eval.go:3398]: 💭 When would `pd` be nil?

**Review by srenatus — COMMENTED** [review]
Looks good so far! 👍

File: topdown/eval.go (hunk 2 above)
Comment (johanfylling) [reply topdown/eval.go:3398]: Probably never. Sometimes I'm guilty of writing a tad too defensive code 😄.

File: ast/term.go (hunk 1 above)
Comment (johanfylling) [reply ast/term.go:560]: Sure, I can move it there 👍

**Review by srenatus — APPROVED** [review]
So if you slept on it and still think it's an OK approach, let's merge it 🥳  Thanks for the quick fix.

**Comment by johanfylling** [comment]
Slept on it. No further epiphanies 😄.

**Pull request merged**